annotation-processors/src/main/kotlin/com/intellij/vim/model/LazyInstance.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2003-2023 The IdeaVim authors
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+package com.intellij.vim.model
+
+import java.lang.invoke.MethodHandles
+import java.lang.invoke.MethodType
+
+open class LazyInstance<T>(private val className: String, private val classLoader: ClassLoader) {
```

Comment:
Why do we extract this class from `LazyVimscriptFunction`?

Reply:
I'm not sure if I understood you correctly. If you're asking "Why do we have `LazyVimscriptFunction` when we have a `LazyInstance`?", then the reason is that VimScript functions have a lateinit name field, which should be handled.

Surely, I could use only the `LazyInstance` and register all the functions in some kind of a map, such as `nameToFunction`. However, it's more convenient to have a class that carries out all the required initialization on its own, rather than doing half of the initialization with the help of `LazyInstance` and manually setting the function name later.